crates/tap-agent/src/test.rs

```diff
@@ -31,6 +58,193 @@ lazy_static! {
         tap_eip712_domain(1, Address::from([0x11u8; 20]),);
 }
 
+pub static PREFIX_ID: AtomicU32 = AtomicU32::new(0);
+
+pub const TRIGGER_VALUE: u128 = 500;
+pub const RECEIPT_LIMIT: u64 = 10000;
+const ESCROW_VALUE: u128 = 1000;
+const BUFFER_DURATION: Duration = Duration::from_millis(100);
+const RETRY_DURATION: Duration = Duration::from_millis(1000);
+const RAV_REQUEST_TIMEOUT: Duration = Duration::from_secs(60);
+const TAP_SENDER_TIMEOUT: Duration = Duration::from_secs(30);
+
+const RAV_REQUEST_BUFFER: Duration = Duration::from_secs(60);
+const ESCROW_POLLING_INTERVAL: Duration = Duration::from_secs(30);
+
+pub fn get_sender_account_config() -> &'static SenderAccountConfig {
+    Box::leak(Box::new(SenderAccountConfig {
+        rav_request_buffer: RAV_REQUEST_BUFFER,
+        max_amount_willing_to_lose_grt: TRIGGER_VALUE + 100,
+        trigger_value: TRIGGER_VALUE,
+        rav_request_timeout: Duration::from_secs(30),
+        rav_request_receipt_limit: 1000,
+        indexer_address: INDEXER.1,
+        escrow_polling_interval: ESCROW_POLLING_INTERVAL,
+        tap_sender_timeout: Duration::from_secs(63),
+    }))
+}
+
+#[allow(clippy::too_many_arguments)]
+#[bon::builder]
+pub async fn create_sender_account(
+    pgpool: PgPool,
+    #[builder(default = HashSet::new())] initial_allocation: HashSet<Address>,
+    #[builder(default = TRIGGER_VALUE)] rav_request_trigger_value: u128,
+    #[builder(default = TRIGGER_VALUE)] max_amount_willing_to_lose_grt: u128,
+    escrow_subgraph_endpoint: Option<&str>,
+    network_subgraph_endpoint: Option<&str>,
+    #[builder(default = RECEIPT_LIMIT)] rav_request_receipt_limit: u64,
+    aggregator_endpoint: Option<Url>,
+) -> (
+    ActorRef<SenderAccountMessage>,
+    Arc<Notify>,
+    String,
+    Sender<EscrowAccounts>,
+) {
+    let config = Box::leak(Box::new(SenderAccountConfig {
+        rav_request_buffer: BUFFER_DURATION,
+        max_amount_willing_to_lose_grt,
+        trigger_value: rav_request_trigger_value,
+        rav_request_timeout: RAV_REQUEST_TIMEOUT,
+        rav_request_receipt_limit,
+        indexer_address: INDEXER.1,
+        escrow_polling_interval: Duration::default(),
+        tap_sender_timeout: TAP_SENDER_TIMEOUT,
+    }));
+
+    let network_subgraph = Box::leak(Box::new(
+        SubgraphClient::new(
+            reqwest::Client::new(),
+            None,
+            DeploymentDetails::for_query_url(
+                network_subgraph_endpoint.unwrap_or(&get_grpc_url().await),
+            )
+            .unwrap(),
+        )
+        .await,
+    ));
+    let escrow_subgraph = Box::leak(Box::new(
+        SubgraphClient::new(
+            reqwest::Client::new(),
+            None,
+            DeploymentDetails::for_query_url(
+                escrow_subgraph_endpoint.unwrap_or(&get_grpc_url().await),
+            )
+            .unwrap(),
+        )
```

Comment:
Same here, get_grpc_url() shouldn't be used here, a simple DUMMY_URL should work.